kong/cmd/utils/prefix_handler.lua

```diff
@@ -202,7 +202,8 @@ local function prepare_prefix(kong_config, nginx_custom_template_path)
   end
 
   -- generate default SSL certs if needed
-  if kong_config.ssl and not kong_config.ssl_cert and not kong_config.ssl_cert_key then
+  if table.concat(kong_config.proxy_listen, ","):find("ssl") and
```

Comment:
Another, more important note: This will also trigger certificate generation if the values looks like so:
```
proxy_listen = ssl-cert-service.com:80
```

If we move the "at least one listen directive has SSL enabled" logic away from the template (as pointed in another comment) and into the Lua-land, we can avoid such surprises.